### PR TITLE
DRY entrypoints

### PR DESCRIPTION
**Pull request by lguychard**
Deduplicates code around extension marker and code intelligence injection in phabricator, integrations & injected script entrypoints.

Also removes comments about extension marker injection being racy in Phabricator and integration entry points: `<script>` tags with the `defer` attribute are executed after the document has been parsed, but before `DOMContentLoaded` fires. Meanwhile, browser extension content scripts are set to run at `document_end`, after `DOMContentLoaded` has fired.

#### Test Plan

- [x] Browser extension
- [x] Bitbucket integration
- [x] Phabricator integration

### CODE REVIEW

File: browser/src/extension/scripts/inject.tsx [hunk 1]
```diff
@@ -43,19 +41,17 @@ async function main(): Promise<void> {
...
-    const extensionMarker = document.createElement('div')
-    extensionMarker.id = 'sourcegraph-app-background'
-    extensionMarker.style.display = 'none'
-    if (document.getElementById(extensionMarker.id)) {
+    // Check if a native integration is already running on the page,
+    // and abort execution if it's the case.
+    // Note: we should not need this check. Integration scripts are run as
+    // script tags with `defer`, they will be fetched and executed before
+    // DOMContentLoaded fires. Meanwhile, the content script is set to run at document_end,
+    // meaning that it will only execute after DOMContentLoaded has fired.
```
Comment (chrismwendt) [inline browser/src/extension/scripts/inject.tsx]: Unless I misunderstand, I'd change "we should not need this check" to something like "this is just a sanity check". I think that would better prepare the reader for the explanation that follows.

File: browser/src/libs/phabricator/extension.tsx [hunk 2]
```diff
@@ -52,7 +28,8 @@ function init(): void {
         // passed the bundle url. Legacy Phabricator extensions inject CSS via the loader.js script
         // so we do not need to do this here.
         if (!window.SOURCEGRAPH_BUNDLE_URL && !window.localStorage.getItem('SOURCEGRAPH_BUNDLE_URL')) {
-            injectModules().catch(err => console.error('Unable to inject modules', err))
+            injectExtensionMarker()
+            injectCodeIntelligence(IS_EXTENSION).catch(err => console.error('Unable to inject code intelligence', err))
```
Comment (felixfbecker) [inline browser/src/libs/phabricator/extension.tsx]: What's the reason to catch this error? I.e. why should we continue if this fails, under what legit reason can it fail?

File: browser/src/libs/sourcegraph/inject.tsx [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+export function signalBrowserExtensionInstalled(): void {
+    injectExtensionMarker()
+
+    window.addEventListener(
+        'load',
+        () => {
+            dispatchSourcegraphEvents()
+        },
+        { once: true }
+    )
```
Comment (felixfbecker) [inline browser/src/libs/sourcegraph/inject.tsx]: I guess this should only happen in the `else` of the `if` below?

File: browser/src/extension/scripts/inject.tsx (hunk 1 above)
Comment (lguychard) [reply browser/src/extension/scripts/inject.tsx]: Good point, thanks! Updated.

File: browser/src/libs/phabricator/extension.tsx (hunk 2 above)
Comment (lguychard) [reply browser/src/libs/phabricator/extension.tsx]: You're right, we shouldn't catch it. Changed in 7890568

File: browser/src/libs/phabricator/extension.tsx [hunk 4]
```diff
@@ -15,70 +13,46 @@ window.SOURCEGRAPH_PHABRICATOR_EXTENSION = true
...
+        return
+    }
+
+    const sourcegraphURL = await getSourcegraphURLFromConduit()
+    const css = await getPhabricatorCSS(sourcegraphURL)
+    const style = document.createElement('style')
+    style.setAttribute('type', 'text/css')
+    style.id = 'sourcegraph-styles'
+    style.textContent = css
+    document.getElementsByTagName('head')[0].appendChild(style)
```
Comment (felixfbecker) [inline browser/src/libs/phabricator/extension.tsx]:
```suggestion
    document.head.appendChild(style)
```

**Pull request merged**